### PR TITLE
Implement BootstrapTour to WebApp homepage

### LINKED ISSUE #508: Add an interactive tutorial
Issue raised by: [developer/user/project requirement]

### What?
Follows on from #416

- In pursuit of a better way of presenting help documentation to users, @ElliotAOram discussed the idea of implementing an interactive tutorial for the WebApp.
- This would allow the user to click through information boxes which highlight and explain various features of the WebApp.
___
- This issue involves:
  * An investigation into what existing packages which can fulfil this requirement.
  * Testing of any potential package(s) - including feasibility testing of whether the time/effort to apply to our WebApp is reasonable.
  * For the most promising package: Implementing a prototype of how we can apply this to the WebApp.
  * If the prototype is successful: Developing the prototype into a full implementation.

### How?
- This issue spawned out of discussions regarding how we can best teach users how to use the Autoreduction WebApp

### How to test the issue is resolved
- Assess whether the interactive tutorial implementation is ready to be deployed

### PR DESCRIPTION
**Pull request by JEThompson97**
### Summary of work
- Bootstrap Tour was identified as an ideal package to provide an interactive tutorial of the webapp.
- This PR implements Bootstrap Tour on the WebApp homepage (overview.html)

image

- Currently the implementation consists of:
  * A `tour.js` file which provides functionality to initialise a tour with specific settings
  * In line JavaScript within `overview.html` to create the tour steps for that page

### How to test your work
- Checkout this branch
- Run the WebApp locally and go to it
- Click the "Tour" button
- Click through the tour by clicking "Next" at each information box

### Additional comments
- To progress with this issue further: **Next steps** are to design the specifics of the tutorial. These include:
  * Design an initial 'storyboard' of the tour steps
  * Write an initial draft of the tutorial copy
  * Discuss & decide whether we'd like a tour-per-page OR a single tour spanning the whole WebApp (OR something in between).

Fixes #508 

**Before merging ensure the release notes have been updated**

### CODE REVIEW

**Comment by JEThompson97** [comment]
@ElliotAOram Ready for review

I've refactored the code to make tours 'stackable':
- For any template 1 or more ` ..._tour.js` files (which are used to hold tour-steps) can be added under `{% block scripts %}` followed by `create_tour.js`
- This will append all steps across the ` ..._tour.js` files in order, and create a tour using this combined list.
- e.g. for _overview.html_, I have added the _overview_tour.js_ steps (showcases elements of _overview.html_), followed by the _navbar_tour.js_ steps

I've used JS files instead of txt (as we originally discussed) as this enables automatic syntax highlighting, which will avoid typo mistakes, and avoid extra work in terms of coming up with a text interpretation-format and interpretation functionality - as the JS files work right out the box.

File: WebApp/autoreduce_webapp/static/javascript/tours/overview-tour.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    {
+        element: "#instrument-btns-container",
+        title: "Instruments",
+        content: "Here are ISIS instruments connected to autoreduction.",
+        placement: "bottom"
+    },
+    {
+        element: "#"+firstInstrumentBtn.id,
+        title: "Instruments",
+        content: "You can click on an instrument to view its reduction jobs.",
```
Comment (ElliotAOram) [inline WebApp/autoreduce_webapp/static/javascript/tours/overview-tour.js:12]: "You can click on an instrument to view its reduction jobs, re-run jobs, pause reduction and set variables for upcoming runs."

**Comment by ElliotAOram** [comment]
Couple of thought to look into:
Is there a way to highlight the element being described? or grey out the others?
We need to find a consistent place for the take tour button on all pages

**Comment by JEThompson97** [comment]
@ElliotAOram 
> Is there a way to highlight the element being described? or grey out the others?

Turns out this was as simple as adding `backdrop: true` to the tour parameters:

image

**Pull request merged**